### PR TITLE
feat: update tailwindcss

### PR DESCRIPTION
**Pull request by ufkaya**
BREAKING CHANGE: update tailwindcss to 2.1.2

As you can see here https://github.com/carforyou/carforyou-components-pkg/pull/341, I think I have some issues updating postcss 8 and that's why I installed the version of tailwindcss that supports postcss 7
I will create another task to update postcss or feel free to help me with my current PR

Note: Before merging this PR will ask QA to test it on Listings and DH

### CODE REVIEW

**Review by lkappeler — COMMENTED** [review]
- I checked the PR you linked but there is no description of the problem, can you elaborate on the issue and the steps you took to try to fix it? (Like I got that there must be some sort of problem with postcssv8 but I don't quite understand what the problem is so far...)

- Do we need to prepare the projects to support this so we are not blocked in making updates on the components or can we install the components library with tailwind v2 

- I assume you followed https://tailwindcss.com/docs/upgrading-to-v2 maybe you can add it to the description so reviews don't need to search for the breaking changes in the new version...

File: src/components/dropdown/menu.tsx [hunk 1]
```diff
@@ -102,7 +102,7 @@ class Menu<T> extends Component<Props<T>> {
         {...getMenuProps(
           {
             className: classNames(
-              "border border-grey-3 absolute z-dropdownMenu bg-white cursor-normal inset-x-0 list-reset scrolling-touch overflow-y-scroll custom-scrollbar max-h-dropdownSM md:max-h-dropdown py-10 shadow-soft rounded-4",
```
[the remark this answers is not in the document]
Comment (ufkaya) [reply src/components/dropdown/menu.tsx:105]: its not supported anymore, https://github.com/tailwindlabs/tailwindcss/pull/2573
We will test it and see if it is still ok

File: package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
   "peerDependencies": {
     "react": ">=16.12.0",
     "react-dom": ">=16.12.0",
-    "tailwindcss": "^1.4.0"
+    "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.1.2"
```
[the remark this answers is not in the document]
Comment (ufkaya) [reply package.json:30]: is telling tailwindcss to use PostCSS 7, they included PostCSS 7 compatibility build

**Comment by ufkaya** [comment]
> * I checked the PR you linked but there is no description of the problem, can you elaborate on the issue and the steps you took to try to fix it? (Like I got that there must be some sort of problem with postcssv8 but I don't quite understand what the problem is so far...)

I also dont know what is the problem, npm ls fails, but I cannot reproduce it on my computer https://app.circleci.com/pipelines/github/carforyou/carforyou-components-pkg/2558/workflows/a6ef542b-92cf-45fb-ac69-66b05f5216dc/jobs/8054
Looks good on my computer and I can even build the pkg

> * Do we need to prepare the projects to support this so we are not blocked in making updates on the components or can we install the components library with tailwind v2

There are some breaking changes that we need to handle, will take care of it

> * I assume you followed https://blog.tailwindcss.com/tailwindcss-v2 maybe you can add it to the description so reviews don't need to search for the breaking changes in the new version...

You can also check the release notes here -> https://github.com/tailwindlabs/tailwindcss/releases/tag/v2.0.0

**Review by masone — APPROVED** [review]
I'm cool with unlocking the tailwind update with the compat package. it's a good isolated step on the upgrade path. I would not want to stop there though and push for postcss8 in the next PR

**Comment by ufkaya** [comment]
https://github.com/carforyou/carforyou-components-pkg/pull/341

**Pull request closed without merging**